src/components/filters/MeasureValueFilter/Dropdown.tsx

```diff
@@ -0,0 +1,121 @@
+// (C) 2019 GoodData Corporation
+import * as React from "react";
+import { injectIntl, InjectedIntlProps } from "react-intl";
+import { IntlWrapper } from "../../core/base/IntlWrapper";
+import { MeasureValueFilterConditionOperator, IValue } from "./types";
+import Overlay from "@gooddata/goodstrap/lib/core/Overlay";
+import { DropdownButton as DefaultDropdownButton } from "./DropdownButton";
+import { DropdownBody } from "./DropdownBody";
+import { getOperatorTranslationKey } from "./utils";
+
+export interface IDropdownButtonProps {
+    isActive?: boolean;
+    onClick: (e: React.SyntheticEvent) => void;
+    measureTitle?: string;
+    operator?: string;
+    operatorTitle?: string;
+    value?: IValue;
+}
+
+export interface IDropdownOwnProps {
+    button?: React.ComponentType<IDropdownButtonProps>;
+    onApply: (operator: MeasureValueFilterConditionOperator, value?: IValue) => void;
+    locale?: string;
+    measureTitle?: string;
+    operator?: MeasureValueFilterConditionOperator;
+    value?: IValue;
+    displayDropdown?: false;
+}
+
+export type IDropdownProps = IDropdownOwnProps & InjectedIntlProps;
+
+interface IDropdownState {
+    displayDropdown: boolean;
+}
+
+class DropdownWrapped extends React.PureComponent<IDropdownProps, IDropdownState> {
+    public static defaultProps: Partial<IDropdownOwnProps> = {
+        button: DefaultDropdownButton,
+        value: {},
+    };
+
+    private toggleButtonRef: EventTarget = null;
+
+    constructor(props: IDropdownProps) {
+        super(props);
+
+        const { displayDropdown } = props;
+
+        this.state = {
+            displayDropdown,
+        };
+    }
+
+    public render() {
+        const { intl, button: DropdownButton, measureTitle, operator, value, locale } = this.props;
+        const { displayDropdown } = this.state;
+
+        const operatorTitle = intl.formatMessage({
+            id: getOperatorTranslationKey(operator ? operator : "ALL"),
+        });
+
+        return (
+            <>
+                <DropdownButton
+                    isActive={displayDropdown}
+                    onClick={this.toggle}
+                    measureTitle={measureTitle}
+                    operatorTitle={operatorTitle}
+                    operator={operator ? operator : "ALL"}
+                    value={value}
+                />
+                {displayDropdown ? (
+                    <Overlay
+                        alignTo={this.toggleButtonRef}
```

Comment:
How about we wrap the `DropdownButton` in some `div` with a well known class and use that class as the argument to `alignTo`? Can that work? It could save us the trouble with the button ref.

Reply:
It is also a way to do this (we actually do it this way with most of our dropdown components). I found inspiration in the implementation of popovers etc. in a few component libraries and I found it better than adding some `div` with unique classname, or id:
https://material-ui.com/components/popover/
https://demos.creative-tim.com/material-kit-react/#/documentation/popovers

Reply:
Ok, fair enough :)